src/components/Table/DataTable.vue

```diff
@@ -0,0 +1,48 @@
+<script setup>
+import { ref, onMounted } from "vue";
+import DataTable from "datatables.net-vue3";
+import DataTablesCore from "datatables.net-dt";
+import "datatables.net-searchpanes-dt";
+
+DataTable.use(DataTablesCore);
+
+const props = defineProps({
+  tableId: String,
+  apiEndpoint: String,
+  columns: Array,
+});
+
+const data = ref([]);
+const BASE_URL = process.env.API_BASE_URL;
+
+onMounted(async () => {
+  try {
+    const response = await fetch(`${BASE_URL}${props.apiEndpoint}`);
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
```

Comment:
One thing we should think about is what we need to show in the UI when there's an HTTP error to support user troubleshooting and analytics.